autogluon/utils/tabular/ml/models/lgb/lgb_model.py

```diff
@@ -144,14 +146,36 @@ def predict_proba(self, X, preprocess=True):
             else:  # Should this ever happen?
                 return y_pred_proba[:, 1]
 
+    def preprocess(self, X, is_train=False):
+        X = super().preprocess(X=X)
+
+        if is_train:
+            requires_internal_map = False
+            for column in X.columns:
+                new_column = re.sub(r'[",:{}[\]]', '', column)
```

Comment:
This is the full set of lGBM-breaking chars?  If not consider adding TODO that there may be others as well

Reply:
This is the code in LGBM that checks, and I've tested many other characters without issue: https://github.com/microsoft/LightGBM/pull/2557/files#diff-c199c652de48f026fa699c5775b6eab1R934

Reply:
non-ASCII are also not supported, but that is probably something we should handle at a higher level, as many models may have issues in that regard.